### PR TITLE
WIP: first start of adding some documentation

### PR DESCRIPTION
**Pull request by JoachimM7**
There should be some code documentation. I would suggest to find a common style for doing this, so I started with adding some documentation. Please feel free to comment.

### CODE REVIEW

File: Sources/MongoKitten/AggregateStage.swift [hunk 1]
```diff
@@ -82,6 +101,44 @@ public struct AggregateBuilderStage {
...
+	///     .unwind(fieldPath: "$newName")
+	/// ])
+	/// ```
+	///
+	/// - Parameters:
+	///   - from: the foreign collection, where the documents will be looked up
+	///   - localField: the name of the field in the input collection that shall match the `foreignField` in the `from` collection
+	///   - foreignField: the name of the field in the `fromCollection` that shall match the `localField` in the input collection
+	///   - newName: the collecting matches will be inserted as an array to the input collection, named as `newName`
+	/// - Returns: returns an `AggregateBuilderStage`
```
Comment (Joannis) [inline Sources/MongoKitten/AggregateStage.swift:141]: Is the first `returns` word in `return an AggregateBuilderStage` added for readability in Xcode? It seems redundant in the `Returns` section.

File: Sources/MongoKitten/AggregateBuilder.swift [hunk 2]
```diff
@@ -52,6 +52,27 @@ public struct AggregateBuilder {
...
+	///
+	/// pipeline.decode(SomeDecodableType.self).forEach { yourStruct in
+	///	    // do sth. with your struct
+	///	}.whenFailure { error in
+	///	    // do sth. with the error
+	/// }
+	/// ```
+	///
+	/// - Parameter build: the `AggregateBuilderStage` as function builders
+	/// - Returns: returns an `AggregateBuilderPipeline`
```
Comment (Joannis) [inline Sources/MongoKitten/AggregateBuilder.swift:75]: I'd add something like:

`.. that should be executed to get results`

Comment (Joannis) [reply Sources/MongoKitten/AggregateBuilder.swift:75]: Maybe not perfect, but I'm wondering if people understand that part. That it's not a sent query yet.

File: Sources/MongoKitten/AggregateStage.swift (hunk 1 above)
Comment (JoachimM7) [reply Sources/MongoKitten/AggregateStage.swift:141]: That's right. Should be just like this?

/// - Returns: An `AggregateBuilderStage`

Comment (Joannis) [reply Sources/MongoKitten/AggregateStage.swift:141]: If you option-click on the function definition or any call to it, your best guess should guide you :)

File: Sources/MongoKitten/AggregateBuilder.swift (hunk 2 above)
Comment (JoachimM7) [reply Sources/MongoKitten/AggregateBuilder.swift:75]: You're right. I will add another aggregation query like lookup

**Comment by Joannis** [comment]
@JoachimM7 is this still work in progress?

**Comment by JoachimM7** [comment]
@Joannis: I would like to add more documentation step by step. Should we merge the PR right now?

**Comment by Joannis** [comment]
Sounds good!

**Pull request merged**